IllegalStateException: Activity did not call finish() prior to onResume() completing

From time to time I experience the following crash.

# Environment
- Samsung Galaxy S8
- Android 8.0

# How to reproduce
1. Uninstall an app by tapping the trash icon.
2. If app is still listed then quickly tap the trash icon again.

# Expected behavior
- App is removed from the listing.

# Observed behavior
- Crash notice `DevDrawer2 (Debug) has stopped`.

# Stacktrace

``` java
java.lang.RuntimeException: Unable to resume activity  {de.psdev.devdrawer.debug/de.psdev.devdrawer.activities.ClickHandlingActivity}: 
 java.lang.IllegalStateException: Activity  {de.psdev.devdrawer.debug/de.psdev.devdrawer.activities.ClickHandlingActivity} did not call finish() prior to onResume() completing
    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3788)
    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3828)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3036)
    at android.app.ActivityThread.-wrap11(Unknown Source:0)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696)
    at android.os.Handler.dispatchMessage(Handler.java:105)
    at android.os.Looper.loop(Looper.java:164)
    at android.app.ActivityThread.main(ActivityThread.java:6938)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
Caused by: java.lang.IllegalStateException: Activity 
 {de.psdev.devdrawer.debug/de.psdev.devdrawer.activities.ClickHandling
 Activity} did not call finish() prior to onResume() completing
    at android.app.Activity.performResume(Activity.java:7356)
    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3763)
    ... 10 more
```

--- a/app/src/main/java/de/psdev/devdrawer/activities/ClickHandlingActivity.kt
+++ b/app/src/main/java/de/psdev/devdrawer/activities/ClickHandlingActivity.kt
@@ -38,7 +38,12 @@ class ClickHandlingActivity: Activity() {
                 Constants.LAUNCH_APP -> startApp(this, packageName)
                 Constants.LAUNCH_APP_DETAILS -> startAppDetails(this, packageName)
                 Constants.LAUNCH_UNINSTALL -> startUninstall(this, packageName)
+                // We need to always call finish
+                else -> finish()
             }
+        } else {
+            // We need to always call finish
+            finish()
         }
     }
 

